src/coreclr/tools/cdac-build-tool/ObjectFileScraper.cs

```diff
@@ -0,0 +1,497 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Diagnostics.DataContract.BuildTool;
+
+public class ObjectFileScraper
+{
+    public static readonly ReadOnlyMemory<byte> MagicLE = new byte[8] { 0x44, 0x41, 0x43, 0x42, 0x4C, 0x4F, 0x42, 0x00 }; // "DACBLOB\0"
+    public static readonly ReadOnlyMemory<byte> MagicBE = new byte[8] { 0x00, 0x42, 0x4F, 0x4C, 0x42, 0x43, 0x41, 0x44 };
+
+    private readonly DataDescriptorModel.Builder _builder;
+
+    public bool Verbose {get;}
+    public ObjectFileScraper(bool verbose, DataDescriptorModel.Builder builder)
+    {
+        Verbose = verbose;
+        _builder = builder;
+    }
+
+    public async Task<bool> ScrapeInput(string inputPath, CancellationToken token)
+    {
+        var bytes = await File.ReadAllBytesAsync(inputPath, token).ConfigureAwait(false);
+        if (!ScraperState.FindMagic(bytes, out var state))
+        {
+            return false;
+        }
+        if (Verbose) {
+            Console.WriteLine($"{inputPath}: magic at {state.MagicStart}");
+        }
+        var header = ReadHeader(state);
+        if (Verbose) {
+            DumpHeaderDirectory(header);
+        }
+        var content = ReadContent(state, header);
+        content.AddToModel(_builder);
+        return true;
+    }
+
+    private sealed class ScraperState
+    {
+        public ReadOnlyMemory<byte> Data { get; }
+        public bool LittleEndian { get; }
+        private long _position;
+
+        public long MagicStart => HeaderStart - MagicLE.Length;
```

Comment:
I assume that both LE and BE are the same length. I would add a comment this is intentional.